packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -755,12 +755,12 @@ class PageTransitionsTheme with Diagnosticable {
   ) {
     TargetPlatform platform = Theme.of(context).platform;
 
-    if (CupertinoRouteTransitionMixin.isPopGestureInProgress(route)) {
+    if (CupertinoRouteTransitionMixin.isPopGestureInProgress(route)) { // The trouble maker
       platform = TargetPlatform.iOS;
```

Comment:
@HansMuller seems like you added this logic in the original issue #21715. Outright removing it seems to fix the linked issue, but I'm assuming it covers an edge case I'm not thinking of?

With the code before in the case of this issue, if the theme did not define a builder for the iOS platform, it was defaulting to the zoom page transition, which I'm guessing didn't play nicely with the intended Cupertino page transitions set. The change below seems to fix it, but I'm tagging you to see if you remember what this code covers and if my fix would break that?

Reply:
I don't remember that hack (from https://github.com/flutter/flutter/pull/21715) however I guess it was possible to end up in a Cupertino pop transition even when the target platform is not iOS and the hack avoided switching to a different transition (based on the actual target platform). 

It looks like someone tried to remove the hack in July but that change had to be reverted due to failing internal tests: https://github.com/flutter/flutter/pull/114303. Plus the change in #114303 was slightly breaking, since it introduced an additional parameter in buildTransitions. That did seem like the right way to handle the problem.




If removing the 5-year-old hack doesn't cause any tests to fail, including all of the internal ones,